File: leetcode_python/1.two-sum.py
class Solution:
    def twoSum(self, nums, target):
        dic = {}
        for i in range(len(nums)):
            if str(target - nums[i]) in dic.keys():
                return [dic[str(target - nums[i])], i]
            if str(nums[i]) not in dic.items():
                dic[str(nums[i])] = i


s = Solution()
s.twoSum([2, 7, 11, 15], 9)

File: leetcode_java/1. two-sum.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int x = nums[i];
            if (map.containsKey(target - x)) {
                return new int[] { map.get(target - x), i };
            }
            map.put(x, i);
        }
        return new int[] { 0, 0 };
    }
}

// class Main {
//     public static void main(String[] args) {
//         // Create a new Solution instance
//         Solution solution = new Solution();
//         // Create a test case
//         int[] nums = new int[] { 2, 3, 7, 11, 15 };
//         // Get the answer
//         int[] answer = solution.twoSum(nums, 9);
//         // Print the answer
//         System.out.print(answer[0]);
//         System.out.print(answer[1]);
//     }
// }